crd-generator/api-v2/src/test/java/io/fabric8/crdv2/generator/v1/JsonSchemaDefaultValueTest.java

```diff
@@ -0,0 +1,175 @@
+
+    @JsonProperty(defaultValue = "13.37d")
+    double defaultValueForDouble;
+  }
+
+  @Test
+  @DisplayName("JsonProperty default value should take precedence over Default annotation")
+  void precedence() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("precedence._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .extracting(JsonNode::asText)
+        .isEqualTo("precedence-from-json-property");
+  }
+
+  @Test
+  @DisplayName("Default annotation for String")
+  void stringFromDefaultAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultAnnotationForString._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .extracting(JsonNode::asText)
+        .isEqualTo("string-from-default-annotation");
+  }
+
+  @Test
+  @DisplayName("JsonProperty defaultValue annotation for boxed Boolean")
+  void booleanBoxedFromJsonPropertyAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultValueForBoxedBoolean._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(BooleanNode.class)
+        .extracting(JsonNode::asBoolean)
+        .isEqualTo(true);
+  }
+
+  @Test
+  @DisplayName("Default annotation for boxed Boolean")
+  void booleanBoxedFromDefaultAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultAnnotationForBoxedBoolean._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(BooleanNode.class)
+        .extracting(JsonNode::asBoolean)
+        .isEqualTo(true);
+  }
+
+  @Test
+  @DisplayName("JsonProperty defaultValue annotation for boolean")
+  void booleanFromJsonPropertyAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultValueForBoolean._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(BooleanNode.class)
+        .extracting(JsonNode::asBoolean)
+        .isEqualTo(true);
+  }
+
+  @Test
+  @DisplayName("Default annotation for boolean")
+  void booleanFromDefaultAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultAnnotationForBoolean._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(BooleanNode.class)
+        .extracting(JsonNode::asBoolean)
+        .isEqualTo(true);
+  }
+
+  @Test
+  @DisplayName("JsonProperty defaultValue annotation for int")
+  void intFromJsonPropertyAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultValueForInt._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(LongNode.class)
```

Comment:
This has to do with the Number parser which considers this to be of long precision.
It really doesn't matter, since what we want is the number to be considered a number and not a String.
Of course, there must be corner cases such as users declaring a field of type in and setting a default value of `999999999999999999999`. My expectation is that the cluster won't accept that CRD.
We could try and consider those corner cases, but if we introduce validations for the default values too, the complexity will be kind of crazy for such a simple feature.

Fixing this is just a matter of using the specific parser for the target type. i.e. adding more if-else clauses.

Reply:
Good enough for me, if we get any complaints over this we should remember to add checks for the boundaries (i.e. `Integer.MAX_VALUE`) etc.